project/plugins/bitbucket/__init__.py

```diff
@@ -15,56 +14,148 @@ def getSecret(configMap):
-             cli_groups = list(thisPermissions.values())
-             break
+    for group in allMembers:
+        if group['name'].lower() in cli_groups:
+            for user in group['members']:
+                if userName == user['username']:
+                    return True
+    return False
 
-     if len(cli_groups) == 0:
-         cli_groups = getGroups(configMap,plugin_tag)
 
-     for group in cli_groups:
-         invGroup = requests.put(
-             "https://api.bitbucket.org/1.0/users/"+configMap['global']['organization']+"/invitations/"
-             + email + "/"+configMap['global']['organization']+"/"+group+ "?access_token=" + access_token)
+def inviteUser(email,configMap,allPermissions, plugin_tag, name):
 
+     done = False
+     userName = email.split('@', 1)[0]
+     #Get Authorization token
+     data = {'grant_type': 'client_credentials'}
 
-     log = 'BitBucket: Email invite sent from Bitbucket.\n'
-     instruction = inviteMessage(configMap,plugin_tag)
-     return getJsonResponse('Bitbucket', email, log, instruction)
+     credentials = requestCredentials(configMap, data)
+     log = errorCheck(credentials, plugin_tag, userName)
+
+     if log:
+         print(log)
+         instruction = log
+     else:
+         my_json = credentials.content.decode('utf8')
+         data = json.loads(my_json)
+         access_token=data.get('access_token')
+
+         cli_groups = []
+         for permission in allPermissions:
+             thisPermissions = ast.literal_eval(permission)
+             if thisPermissions['plugin'] == plugin_tag:
+                 del thisPermissions['plugin']
+                 cli_groups = list(thisPermissions.values())
+                 break
+
+         if len(cli_groups) == 0:
+             cli_groups = getGroups(configMap,plugin_tag)
+
+         #get all groups
+         groups=requests.get("https://api.bitbucket.org/1.0/groups/"+configMap['global']['organization']+"?access_token="+access_token)
+         my_json = groups.content.decode('utf8')
+         allMembers = json.loads(my_json)
+
+         if userExists(allMembers, userName, cli_groups):
+             log  = ("The user " + userName + " cannot be invited. The user already exists")
+             instruction = log
+             print(log)
+
+         else:
+             for group in cli_groups:
+                    invGroup = requests.put(
+                         "https://api.bitbucket.org/1.0/users/"+configMap['global']['organization']+"/invitations/"
+                         + email + "/"+configMap['global']['organization']+"/"+group.lower() + "?access_token=" + access_token)
+
+             log = errorCheck(invGroup, plugin_tag, userName)
+
+             if log:
+                print(log)
+                instruction = log
+             else:
+                log = 'BitBucket: Email invite sent from Bitbucket.\n'
+                instruction = inviteMessage(configMap,plugin_tag)
+                done = True
```

Comment:
Does this handle the case where there are multiple groups specified in cli_groups?